examples/3d/motion_blur.rs

```diff
@@ -0,0 +1,401 @@
+        transform.translation.x = race_track_shape(0.0, t).x;
+        transform.translation.z = race_track_shape(0.0, t).y;
+        transform.translation.y = -0.59;
+        let delta = transform.translation - prev;
+        transform.look_to(delta, Vec3::Y);
+        for child in children.iter() {
+            let Ok(mut wheel) = spins.get_mut(*child) else {
+                continue;
+            };
+            let radius = wheel.scale.x;
+            let circumference = 2.0 * std::f32::consts::PI * radius;
+            let angle = delta.length() / circumference * std::f32::consts::PI * 2.0;
+            wheel.rotate_local_y(angle);
+        }
+    }
+}
+
+fn move_camera(
+    mut camera: Query<(&mut Transform, &mut Projection), Without<CameraTracked>>,
+    tracked: Query<&Transform, With<CameraTracked>>,
+    mode: Res<CameraMode>,
+) {
+    let tracked = tracked.single();
+    let (mut transform, mut projection) = camera.single_mut();
+    match *mode {
+        CameraMode::Track => {
+            transform.look_at(tracked.translation, Vec3::Y);
+            transform.translation = Vec3::new(15.0, 1.0, 0.0);
+            if let Projection::Perspective(perspective) = &mut *projection {
+                perspective.fov = 0.05;
+            }
+        }
+        CameraMode::Chase => {
+            transform.translation =
+                tracked.translation + Vec3::new(0.0, 0.15, 0.0) + tracked.back() * 0.6;
+            transform.look_to(*tracked.forward(), Vec3::Y);
+            if let Projection::Perspective(perspective) = &mut *projection {
+                perspective.fov = 1.0;
+            }
+        }
+    }
+}
+
+fn uv_debug_texture() -> Image {
+    use bevy::render::texture::{ImageAddressMode, ImageSampler, ImageSamplerDescriptor};
+    const TEXTURE_SIZE: usize = 7;
+
+    let mut palette = [
+        164, 164, 164, 255, 168, 168, 168, 255, 153, 153, 153, 255, 139, 139, 139, 255, 153, 153,
+        153, 255, 177, 177, 177, 255, 159, 159, 159, 255,
+    ];
+
+    let mut texture_data = [0; TEXTURE_SIZE * TEXTURE_SIZE * 4];
+    for y in 0..TEXTURE_SIZE {
+        let offset = TEXTURE_SIZE * y * 4;
+        texture_data[offset..(offset + TEXTURE_SIZE * 4)].copy_from_slice(&palette);
+        palette.rotate_right(12);
+    }
+
+    let mut img = Image::new_fill(
+        bevy::render::render_resource::Extent3d {
+            width: TEXTURE_SIZE as u32,
+            height: TEXTURE_SIZE as u32,
+            depth_or_array_layers: 1,
+        },
+        bevy::render::render_resource::TextureDimension::D2,
+        &texture_data,
+        bevy::render::render_resource::TextureFormat::Rgba8UnormSrgb,
+        bevy::render::render_asset::RenderAssetUsages::RENDER_WORLD,
+    );
+    img.sampler = ImageSampler::Descriptor(ImageSamplerDescriptor {
+        address_mode_u: ImageAddressMode::Repeat,
+        address_mode_v: ImageAddressMode::MirrorRepeat,
+        mag_filter: bevy::render::texture::ImageFilterMode::Nearest,
```

Comment:
nit: Is there a reason why you imported `ImageSampler` and `ImageAddressMode` but not `ImageFilterMode`/`TextureDimension`/`Extent3d`/`TextureFormat`/`RenderAssetUsages`?

Reply:
Ah, I could move the imports to the top of the function. Should clean that up.

Reply:
I would just use glob imports for wgpu stuff, if you have more than ~7.